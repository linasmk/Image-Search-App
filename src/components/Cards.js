/* ========= App Dependencies ============= */
import React, { useState } from "react";
import PropTypes from "prop-types";
/* ========= Components =============== */
import CardModal from "@comp/CardModal";
import Loader from "@comp/Loader";
/* ========= Code ============= */

export const Cards = ({ images, topLoader, noResultsError }) => {
  const [cardModalId, setCardModalId] = useState(undefined);
  const openCardModal = (id) => setCardModalId(id);
  const closeCardModal = () => setCardModalId(undefined);

  return (
    <div>
      {topLoader ? (
        <Loader display="block" />
      ) : (
        <section className="card-list">
          {noResultsError ? (
            <section className="no-results">
              <h3 className="no-results__warning">
                Your search keyword did not return any results!
              </h3>
            </section>
          ) : (
            images.map((image) => (
              <div className="card" key={image.id}>
                <img
                  role="presentation"
                  className="card--image"
                  onClick={() => openCardModal(image.id)}
                  alt={image.alt_description}
                  src={image.urls.small}
                  width="50%"
                  height="50%"
                />
                <CardModal
                  openCardModal={cardModalId === image.id}
                  closeCardModal={closeCardModal}
                  altDescription={image.alt_description}
                  imageID={image.id}
                  likes={image.likes}
                  urlsFull={image.urls.full}
                  firstName={image.user.first_name}
                  lastName={image.user.last_name}
                  download={image.links.download}
                />
              </div>
            ))
          )}
        </section>
      )}
    </div>
  );
};
Cards.defaultProps = {
  images: undefined,
  topLoader: false,
  noResultsError: false
};
Cards.propTypes = {
  images: PropTypes.arrayOf(PropTypes.object),
  topLoader: PropTypes.bool,
  noResultsError: PropTypes.bool
};
export default Cards;
