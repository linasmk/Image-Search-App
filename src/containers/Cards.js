/* ===== Redux ===== */
import { connect } from "react-redux";

/* ====== Components ===== */
import CardsComponent from "@comp/Cards";

/* ========= Code ========= */
// export const Cards = connect((state) => ({
//   images: state.images
// }))(CardsComponent);
