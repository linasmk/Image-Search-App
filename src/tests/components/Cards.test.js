/* ========= App Dependencies ============= */
import React from "react";
import { shallow } from "enzyme";

/* ========= Components ============= */
import { Cards } from "@comp/Cards";
import images from "@tests/fixtures/images";
/* ========= Code ============= */

it("renders Cards component with images array correctly", () => {
  const wrapper = shallow(<Cards images={images} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders Cards component with with empyt images array correctly", () => {
  const wrapper = shallow(<Cards images={[]} />);
  expect(wrapper).toMatchSnapshot();
});
