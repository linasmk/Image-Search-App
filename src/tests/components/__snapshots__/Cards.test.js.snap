// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Cards component with images array correctly 1`] = `
<div>
  <section
    className="card-list"
  >
    <div
      className="card"
      key="1"
    >
      <img
        className="card--image"
        height="50%"
        onClick={[Function]}
        role="presentation"
        src="https://images.unsplash.com/photo-1552053831-71594a27632d?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxNzc2NzZ8MHwxfHNlYXJjaHwxfHxkb2d8ZW58MHx8fA&ixlib=rb-1.2.1&q=80&w=400"
        width="50%"
      />
      <CardModal
        altDescription={null}
        closeCardModal={[Function]}
        download="https://unsplash.com/photos/Sg3XwuEpybU/download"
        firstName="Martin"
        imageID="1"
        lastName="Weaver"
        likes={505}
        openCardModal={false}
        urlsFull="https://images.unsplash.com/photo-1552053831-71594a27632d?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxNzc2NzZ8MHwxfHNlYXJjaHwxfHxkb2d8ZW58MHx8fA&ixlib=rb-1.2.1&q=85"
      />
    </div>
    <div
      className="card"
      key="2"
    >
      <img
        className="card--image"
        height="50%"
        onClick={[Function]}
        role="presentation"
        src="https://images.unsplash.com/photo-1568595502714-5a8aa68c5fd7?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxNzc2NzZ8MHwxfHNlYXJjaHwxfHxsZW1vbnxlbnwwfHx8&ixlib=rb-1.2.1&q=80&w=400"
        width="50%"
      />
      <CardModal
        altDescription={null}
        closeCardModal={[Function]}
        download="https://unsplash.com/photos/tP6ETrtLBWA/download"
        firstName={null}
        imageID="2"
        lastName="Stalken"
        likes={600}
        openCardModal={false}
        urlsFull="https://images.unsplash.com/photo-1568595502714-5a8aa68c5fd7?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxNzc2NzZ8MHwxfHNlYXJjaHwxfHxsZW1vbnxlbnwwfHx8&ixlib=rb-1.2.1&q=85"
      />
    </div>
    <div
      className="card"
      key="3"
    >
      <img
        className="card--image"
        height="50%"
        onClick={[Function]}
        role="presentation"
        src="https://images.unsplash.com/photo-1517842645767-c639042777db?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxNzc2NzZ8MHwxfHNlYXJjaHwxfHxub3Rlc3xlbnwwfHx8&ixlib=rb-1.2.1&q=80&w=400"
        width="50%"
      />
      <CardModal
        altDescription={null}
        closeCardModal={[Function]}
        download="https://unsplash.com/photos/_HtKNv8zs5s/download"
        firstName="Robin"
        imageID="3"
        lastName="Tarver"
        likes={900}
        openCardModal={false}
        urlsFull="https://images.unsplash.com/photo-1579208581155-feeb3bbb4e60?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxNzc2NzZ8MHwxfHNlYXJjaHwxfHxub3RlfGVufDB8fHw&ixlib=rb-1.2.1&q=85"
      />
    </div>
  </section>
</div>
`;

exports[`renders Cards component with with empyt images array correctly 1`] = `
<div>
  <section
    className="card-list"
  />
</div>
`;
